webui/react/src/components/NotebookModal.tsx

```diff
@@ -0,0 +1,234 @@
+import { Button, Col, InputNumber, Modal, Row } from 'antd';
+import { Form, Input, Select } from 'antd';
+import { ModalProps } from 'antd/es/modal/Modal';
+import yaml from 'js-yaml';
+import React, { useCallback, useEffect, useState } from 'react';
+import MonacoEditor from 'react-monaco-editor';
```

Comment:
**suggestion:** take a look at `ExperimentInfoBox.tsx` on how `monaco-editor` gets imported via `React.lazy`. This is because `monaco-editor` is a really really big vendor package in terms of bundle size and we only want to load it when absolutely necessary.
```
const MonacoEditor = React.lazy(() => import('react-monaco-editor'));

...

// in JSX
        <React.Suspense
          fallback={<div className={css.loading}><Spinner className="minHeight" /></div>}>
          <MonacoEditor {...props} />
        </React.Suspense>
```
`React.Suspense` will basically show a loader while our WebUI loads a separate bundle for `monaco-editor`